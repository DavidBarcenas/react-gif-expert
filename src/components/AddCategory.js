import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [value, setValue] = useState('');

  const handleValueChange = (e) => setValue(e.target.value);
  const handlesubmit = (e) => {
    e.preventDefault();
    if (value.trim().length > 2) {
      // setCategories((cats) => [...cats, value]);
      setCategories((cats) => [value]);
      setValue('');
    }
  };
  return (
    <form onSubmit={handlesubmit}>
      <input
        type="text"
        placeholder="Search..."
        value={value}
        onChange={handleValueChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
