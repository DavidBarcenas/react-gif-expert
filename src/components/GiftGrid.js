import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <div>
      <h3 className="category__subtitle">{category}</h3>
      {loading && <p>Loading...</p>}
      <ul className="Gifs__list fadeInUp">
        {images.map((img) => (
          <li key={img.id}>
            <img src={img.url} alt={img.title} />
          </li>
        ))}
      </ul>
    </div>
  );
};
