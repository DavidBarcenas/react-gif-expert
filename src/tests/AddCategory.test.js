import React from 'react';
const { shallow } = require('enzyme');
import { AddCategory } from '../components/AddCategory';

describe('AddCategory testing', () => {
  const setCategorys = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategorys} />);

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change input value', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value } });
  });
});
