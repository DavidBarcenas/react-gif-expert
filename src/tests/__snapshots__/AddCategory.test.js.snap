// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory testing should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
